packages/beacon-node/src/chain/validation/dataColumnSidecar.ts

```diff
@@ -4,51 +4,164 @@ import {
+  // 7) [REJECT] The sidecar's block's parent passes validation.
+  const blockState = await chain.regen
+    .getBlockSlotState(parentRoot, blockHeader.slot, {dontTransferCache: true}, RegenCaller.validateGossipBlock)
+    .catch(() => {
+      throw new DataColumnSidecarGossipError(GossipAction.IGNORE, {
+        code: DataColumnSidecarErrorCode.PARENT_UNKNOWN,
+        parentRoot,
+      });
+    });
+
+  // 5) [REJECT] The proposer signature of sidecar.signed_block_header, is valid with respect to the block_header.proposer_index pubkey.
+  const signatureSet = getBlockHeaderProposerSignatureSet(blockState, dataColumnSidecar.signedBlockHeader);
+  // Don't batch so verification is not delayed
+  // TODO: (@matthewkeil) Should this and the blob signature be done main thread like the block?  Talk with @twoeths
+  if (!(await chain.bls.verifySignatureSets([signatureSet], {verifyOnMainThread: true}))) {
+    throw new DataColumnSidecarGossipError(GossipAction.REJECT, {
+      code: DataColumnSidecarErrorCode.PROPOSAL_SIGNATURE_INVALID,
     });
   }
 
-  if (!validateInclusionProof(dataColumnSideCar, chain.metrics)) {
+  // 9) [REJECT] The current finalized_checkpoint is an ancestor of the sidecar's block
+  //             -- i.e. get_checkpoint_block(store, block_header.parent_root, store.finalized_checkpoint.epoch)
+  //                     == store.finalized_checkpoint.root
+  // Handled by 7)
+
+  // 10) [REJECT] The sidecar's kzg_commitments field inclusion proof is valid as verified by
+  //              verify_data_column_sidecar_inclusion_proof
+  //              TODO: Can cache result on (commitments, proof, header) in the future
+  if (!verifyDataColumnSidecarInclusionProof(dataColumnSidecar)) {
     throw new DataColumnSidecarGossipError(GossipAction.REJECT, {
       code: DataColumnSidecarErrorCode.INCLUSION_PROOF_INVALID,
-      slot: dataColumnSideCar.signedBlockHeader.message.slot,
-      columnIdx: dataColumnSideCar.index,
+      slot: dataColumnSidecar.signedBlockHeader.message.slot,
+      columnIdx: dataColumnSidecar.index,
+    });
+  }
+
+  // 11) [REJECT] The sidecar's column data is valid as verified by verify_data_column_sidecar_kzg_proofs
+  try {
+    verifyDataColumnSidecarKzgProofs(
+      dataColumnSidecar.kzgCommitments,
+      Array.from({length: dataColumnSidecar.column.length}, () => dataColumnSidecar.index),
+      dataColumnSidecar.column,
+      dataColumnSidecar.kzgProofs
+    );
+  } catch {
+    throw new DataColumnSidecarGossipError(GossipAction.REJECT, {
+      code: DataColumnSidecarErrorCode.INVALID_KZG_PROOF,
+      slot: blockHeader.slot,
+      columnIdx: dataColumnSidecar.index,
+    });
+  }
+
+  // 12) [IGNORE] The sidecar is the first sidecar for the tuple (block_header.slot, block_header.proposer_index,
+  //              sidecar.index) with valid header signature, sidecar inclusion proof, and kzg proof
```

Comment:
Not sure how best to implement this. Sounds like if you receive the same sidecar twice (identified by the tuple), you can ignore it.

Reply:
I added a `hasDataColumnSidecar` to seenGossipBlockInput and updated `getGossipBlockInput` to throw an IGNORE if it's already been seen. If we wanted we could move the call into the data column validation and call `chain.seenGossipBlockInput.hasDataColumnSidecar` instead

Reply:
there is an issue with this because we first add to our seencache and then go gossip validations (even for blobs) but since we used to overwrite there it was less of concern.

however the right solution defintely is to do gossip validation before adding in seenblock cache which can be done in the followup PR i guess